### PR TITLE
Issue #39820 fixed warnings in com_contact

### PR DESCRIPTION
**Pull request by GaryA**
Pull Request for Issue #39820  .

### Summary of Changes
Line 85 of components\com_contact\tmpl\category\default_items.php
Remove test for item is owned by current user when generating table headings, since there is no current item at this point

### Testing Instructions
Create some contacts and enable users to edit their own contact details in the front end.
Enable PHP warnings.
Go to the contacts list view

### Actual result BEFORE applying this Pull Request
Everything works but 2 PHP warnings are generated about an undefined variable

### Expected result AFTER applying this Pull Request
Everything works exactly the same but there are no warnings

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_contact/tmpl/category/default_items.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ class="inputbox" onchange="document.adminForm.submit();"
                             <th scope="col">
                                 <?php echo Text::_('COM_CONTACT_CONTACT_DETAILS'); ?>
                             </th>
-                            <?php if ($canEdit || ($canDo->get('core.edit.own') && $item->created_by === $userId)) : ?>
+                            <?php if ($canEdit || ($canDo->get('core.edit.own'))) : ?>
```
Comment (Quy) [inline components/com_contact/tmpl/category/default_items.php]:
```suggestion
                            <?php if ($canEdit || $canDo->get('core.edit.own')) : ?>
```

**Comment by joomdonation** [comment]
While this PR addresses the warning issue, it will hide the potential logical error. Think about the following case :

- User has `core.edit.own` permission but does not have `core.edit` permission.
- The current list of contacts does not have any contact belongs to him

With the proposed code, the Edit column will still being displayed but there is no contact items in the table editable, and it is not right. We could use the code like below handle it:

```php
$showEditColumn = false;

if ($canEdit) {
    $showEditColumn = true;
} elseif ($canDo->get('core.edit.own') && !empty($this->items)) {
    foreach ($this->items as $item) {
        if ($item->created_by == $userId) {
            $showEditColumn = true;
            break;
        }
    }
}
```

After that blog of code, $showEditColumn will determine if we should show edit column. I would even go further by adding $showEditColumn as a property of view class and move that logical calculation to the view to avoid too much logic handle inside the layout.

**Comment by GaryA** [comment]
Tried to add $showEditColumn to the view class but couldn't get it to work (not very familiar with Joomla code)
It doesn't really save much since the layout has to calculate for each row anyway, using the same logic.

File: components/com_contact/tmpl/category/default_items.php [hunk 2]
```diff
@@ -28,6 +28,18 @@
...
+if ($canEdit) {
+	$showEditColumn = true;
+} elseif ($canDo->get('core.edit.own') && !empty($this->items)) {
+	foreach ($this->items as $item) {
+		if ($item->created_by == $userId) {
+			$showEditColumn = true;
+			break;
+		}
+	}
+}
```
Comment (richard67) [inline components/com_contact/tmpl/category/default_items.php]:
```suggestion
    $showEditColumn = true;
} elseif ($canDo->get('core.edit.own') && !empty($this->items)) {
    foreach ($this->items as $item) {
        if ($item->created_by == $userId) {
            $showEditColumn = true;
            break;
        }
    }
}
```
Fix failing code style check in drone reported here https://ci.joomla.org/joomla/joomla-cms/61738/1/6 :
- Use spaces and not tabs for indentation
- White space (tab) at the end of line 41

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 2da47f20942b51d3f35ab5abee3af1636d63b143<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39826">issues.joomla.org/tracker/joomla-cms/39826</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 2da47f20942b51d3f35ab5abee3af1636d63b143<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39826">issues.joomla.org/tracker/joomla-cms/39826</a>.</sub>

**Comment by joomdonation** [comment]
RTC. The current change is fine as it is. Further improvement if required could be done in separate PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39826">issues.joomla.org/tracker/joomla-cms/39826</a>.</sub>

**Comment by roland-d** [comment]
Thank you

**Pull request merged**